api/src/services/listing.service.ts

```diff
@@ -1618,4 +1622,120 @@ export class ListingService implements OnModuleInit {
 
     return listing.jurisdictionId;
   }
+
+  async lotteryStatus(
+    dto: ListingLotteryStatus,
+    requestingUser: User,
+  ): Promise<SuccessDTO> {
+    const storedListing = await this.findOrThrow(
+      dto.listingId,
+      ListingViews.details,
+    );
+
+    await this.permissionService.canOrThrow(
+      requestingUser,
+      'listing',
+      permissionActions.update,
+      {
+        id: storedListing.id,
+        jurisdictionId: storedListing.jurisdictionId,
+      },
+    );
+
+    if (storedListing.status !== ListingsStatusEnum.closed) {
+      throw new BadRequestException(
+        'Lottery status cannot be changed until listing is closed.',
+      );
+    }
+
+    const isAdmin = requestingUser.userRoles?.isAdmin;
+    const currentStatus = storedListing.lotteryStatus;
+
+    // TODO: remove when all status logic has been implemented
+    let res;
+
+    switch (dto?.lotteryStatus) {
+      case LotteryStatusEnum.ran: {
+        if (!isAdmin) {
+          throw new ForbiddenException();
+        } else if (
+          currentStatus === LotteryStatusEnum.approved ||
+          currentStatus === LotteryStatusEnum.releasedToPartners ||
+          currentStatus === LotteryStatusEnum.publishedToPublic
+        ) {
+          // TODO: add retracted to history
+          // TODO: remove when all status logic has been implemented
+          res = await this.prisma.listings.update({
+            data: {
+              lotteryStatus: dto?.lotteryStatus,
+            },
+            where: {
+              id: dto.listingId,
+            },
+          });
+        } else {
+          // TODO: add ran to history
+        }
+        break;
+      }
+      case LotteryStatusEnum.errored: {
+        // TODO
+        break;
+      }
```

Comment:
My guess is this status wont be changed via the endpoint but rather during the lottery run function. But it's cool to keep it here as a placeholder here for now

Reply:
I figure we can use the service function, not the endpoint, for wherever the lottery is run to set `errored` and `ran` but it can always be broken out later if we want to handle it a different way